src/TMS/dev/HeatPump.cpp

```diff
@@ -0,0 +1,36 @@
+#include <TMS/dev/HeatPump.hpp>
+
+namespace TMS {
+
+HeatPump::HeatPump(IO::PWM& pwm, uint8_t initSpeed) : pwm(pwm) {
+    // Ensures initSpeed falls in the required range
+    if (initSpeed < MIN_INIT_SPEED) {
+        this->initSpeed = MIN_INIT_SPEED;
+    } else if (initSpeed > MAX_SPEED) {
+        this->initSpeed = MAX_SPEED;
+    } else {
+        this->initSpeed = initSpeed;
+    }
```

Comment:
MIN_INIT_SPEED is 7.  But initSpeed is supposed to be ranged from 0-100.  Shouldn't this check if initSpeed < 0 here.

Reply:
The MIN_INIT_SPEED had to be 7 to make sure the duty cycle and period result in an impulse of >= 3 ms. I changed some things around and added documentation to make this clearer. Let me know if anything is still vague.